### ISSUE TITLE
[P1] #if character literal \x hex escapes and multi-char constants produce wrong values

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#134]
## Problem

The preprocessor's `#if` expression evaluator in `src/frontend/preprocessor/conditionals.rs` (lines 345-383) has two bugs in character literal parsing:

### Bug 1: \x hex escapes not handled

The escape handler recognizes `\n`, `\t`, `\0`, etc. but has no case for `\x` hex escapes. The `\x` falls through to the `other => other` arm and produces the value of `x` (120) instead of the hex value.

```c
#if '\x41' == 65
int correct;  // GCC: included. CCC: EXCLUDED (evaluates to 120, not 65)
#endif
```

Also missing: multi-digit octal escapes. `'\012'` only consumes `\0` (value 0), leaving `12` as stray tokens.

### Bug 2: Multi-character constants not handled

GCC/Clang support multi-character constants in `#if` expressions (e.g., `'AB'`, `'ABCD'`). CCC reads only the first character and skips to the closing quote.

```c
#if 'AB' == 0x4142  // or 0x4241 depending on endianness
int fourcc;  // GCC: included. CCC: EXCLUDED ('AB' evaluates to just 65)
#endif
```

### Impact

Both bugs silently produce wrong conditional compilation decisions — entire blocks of code are incorrectly included or excluded with no diagnostic.

## Files to modify

- `src/frontend/preprocessor/conditionals.rs` — `tokenize_expr()` character literal parser (~line 345)

## Acceptance criteria

- [ ] `'\x41'` evaluates to 65 in `#if` expressions
- [ ] `'\xff'` evaluates to 255
- [ ] `'\012'` evaluates to 10 (octal)
- [ ] Multi-character constants produce implementation-defined values matching GCC behavior
- [ ] `cargo test --lib` passes
- [ ] New tests for hex escapes, octal escapes, and multi-char constants in `#if`
